src/pages/SetPasswordPage.js

```diff
@@ -93,20 +84,11 @@ class SetPasswordPage extends Component {
             <SafeAreaView style={[styles.signInPage]}>
                 <SignInPageLayout>
                     <View style={[styles.mb4]}>
-                        <Text style={[styles.formLabel]}>
-                            {this.props.translate('setPasswordPage.enterPassword')}
-                        </Text>
-                        <TextInput
-                            style={[styles.textInput]}
-                            value={this.state.password}
-                            secureTextEntry
-                            autoCompleteType="password"
-                            textContentType="password"
-                            onChangeText={text => this.setState({password: text})}
+                        <NewPasswordForm
+                            password={this.state.password}
+                            updatePassword={password => this.setState({password})}
+                            updateIsFormValid={isValid => this.setState({isFormValid: isValid})}
```

Comment:
My bad on not catching this on the first review, but since we're only passing back up data to validate it in the `validateAndSubmitForm()` method, it makes more sense to contain all of that in `NewPasswordForm` to prevent unnecessary re-renders of SetPasswordPage

Here's what I suggest doing:

1. Remove the password and the isFormValid from the state in SetPasswordPage and remove the update methods from the props
2. Add those items to NewPasswordForm
3. In NewPasswordForm, create a new method that will validate the password and then call the props.onSubmitEdit method
4. Call that new method on the onSubmitEditing prop in TextInput inside NewPasswordForm.js

Reply:
This is fine by me but the submit button is outside this form fragment. That submit function inside is triggerable via the return key or some native done button etc.

The reason I left the submit button outside is I imagined this would be re-usable in the change password page in the settings RHS where there is another field in the form (the current password field) and the state for that would be outside this component.

Reply:
Ah I see what you mean, feel free to disregard this comment then 👍